backend/checkmach.ml

```diff
@@ -469,17 +469,23 @@ end = struct
     let l =
       List.concat [f div "diverge"; f nor ""; f exn "exceptional return"]
     in
-    let cutoff = !Flambda_backend_flags.checkmach_details_cutoff in
     let len = List.length l in
-    if cutoff < 0 || len < cutoff
-    then l
-    else if cutoff = 0
-    then (* don't even print the dots *)
+    match !Flambda_backend_flags.checkmach_details_cutoff with
+    | Keep_all -> l
+    | No_details ->
+      (* don't even print the dots *)
+      assert (len = 0);
```

Comment:
I am not sure why the condition is always `true`.
Is it because `Witnesses.elements` in `f` above
has to return an empty collection because its
result is also controlled by the cutoff flag?

Reply:
yes